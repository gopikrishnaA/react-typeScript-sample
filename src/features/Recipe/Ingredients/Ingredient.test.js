import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import Ingredient from "./Ingredient";

describe("Ingredient", () => {
  let wrapper;

  let props = {
    ingredient: {
      name: 'test',
      measures: {
        us: {
          amount: 1000,
          unitShort: '$'
        }
      }
    },
    clicked: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<Ingredient {...props} />);
  });

  describe("on start", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(wrapper.find(".d-flex"))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("should click on  particular ingredient", () => {
      wrapper.find(".d-flex").simulate('click');
      expect(wrapper.find(".d-flex").props().children).toContain(1000);
    })
  });
});
