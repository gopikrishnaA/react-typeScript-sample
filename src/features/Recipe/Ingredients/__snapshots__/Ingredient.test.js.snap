// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient on start renders correctly 1`] = `
<div
  className="d-flex align-items-center p-1 m-1"
  onClick={[Function]}
>
  <i
    className="fas fa-cart-arrow-down mr-2 d-inline-block"
    title="Add to Shop List"
  />
  1000
   
  $
   
  test
</div>
`;
